x-pack/plugins/core_telemetry/server/lib/collectors/visualizations/get_usage_collector.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get } from 'lodash';
+import { IHapiServer } from '../../../../';
+import { PLUGIN_ID, VIS_TELEMETRY_TASK, VIS_USAGE_TYPE } from '../../../../constants';
+
+export function getUsageCollector(server: IHapiServer) {
+  const { taskManager } = server;
+  return {
+    type: VIS_USAGE_TYPE,
+    fetch: async () => {
+      let docs;
+      try {
+        ({ docs } = await taskManager.fetch({
+          bool: { filter: { term: { _id: `${PLUGIN_ID}-${VIS_TELEMETRY_TASK}` } } },
+        }));
+      } catch (err) {
+        if (err.constructor === Error && err.toString().indexOf('Error: NotInitialized') === 0) {
```

Comment:
Why is this fine? Might be nice to comment it. I'm assuming it's because this might run before the task manager index is ever created? It's not clear from the code. Also, is there a way to do this test without relying on a consistent `toString()` representation of the error? This check feels brittle, but there may not be a better way.

Reply:
ddb96858b6

Reply:
I tried making our check easier to rely on